cdhweb/people/exodus.py

```diff
@@ -43,3 +48,53 @@ def people_exodus():
         elif profile.thumb:
             person.image = get_wagtail_image(profile.thumb)
             person.save()
+
+
+def user_group_exodus():
+    """Convert groups from Mezzanine configuration to Wagtail configuration.
+
+    - Removes any groups outside Wagtail default Moderators and Editors
+    - Removes all users with the is_staff flag set to False
+    - Removes all users with the is_active flag set to False
+    - Moves all remaining users into the Editors group
+
+    Historically, many users were given the is_superuser flag to avoid a bug
+    with Mezzanine permissions. In Wagtail, this is no longer necessary. In
+    addition, there are two new default groups: Moderators and Editors.
+
+    Users with is_staff or is_active set to False (preventing login) will be 
+    deleted, since there's no need to keep them in the new system. User accounts
+    are no longer associated with content (instead, People are).
+
+    Note that you will still need to assign users to the Moderators group and
+    turn on is_superuser manually after this runs. This can be done in console.
+    """
+
+    # get models and error if any aren't present - they should be since
+    # wagtail's own migrations have already run
+    User = get_user_model()
+    moderators = Group.objects.get(name="Moderators")
+    editors = Group.objects.get(name="Editors")
+
+    # remove any groups that aren't editors or moderators - in cdhweb 2.x we
+    # had an unused "Content Editor" group that should be deleted
+    Group.objects.exclude(pk__in=(moderators.pk, editors.pk)).delete()
+
+    # remove any users who had login disabled for some reason, except the
+    # script user
+    non_staff = 0
+    for user in User.objects.filter(Q(is_staff=False) | Q(is_active=False)) \
+                            .exclude(settings["SCRIPT_USERNAME"]):
+        non_staff += 1
+        logging.debug("removing user %s with is_staff=False" % user)
+        user.delete()
+    if non_staff:
+        logging.info("removed %d non-staff users" % non_staff)
+
+    # assign all remaining users to editors group; remove superuser.
+    for user in User.objects.all():
+        logging.debug("adding user %s to wagtail editors" % user)
+        editors.user_set.add(user)
+        user.is_superuser = False
+        user.save()
```

Comment:
Not that it matters for efficiency with the number of records we have, but you could use a queryset `update()` to set `is_superuser=False` on all users in a single line. I forget the syntax, but imagine there's a way to add a list of users to a group all at once.

Reply:
gonna keep it this way so that we can still log each user added, if that's ok